src/world/Actor/BNpc.cpp

```diff
@@ -521,7 +522,12 @@ void Sapphire::Entity::BNpc::update( int64_t currTime )
         }
 
         if( distance > minActorDistance )
-          moveTo( pHatedActor->getPos() );
+        {
+          uint64_t tick = Util::getTimeMs();
+          auto pTeriMgr = m_pFw->get< World::Manager::TerritoryMgr >();
+          if ( ( tick - m_lastAttack ) > 600 && pTeriMgr->isDefaultTerritory( getCurrentZone()->getTerritoryTypeId() ) )
```

Comment:
is there a particular reason for this check? not sure why this is needed

Reply:
checked retail, every bnpcs stays a few after auto attack

Reply:
no I mean the `pTeriMgr->isDefaultTerritory( getCurrentZone()->getTerritoryTypeId() )`

Reply:
I'm not sure about raid bosses or dungeons

Reply:
hmmm, alright, maybe leave it out for now then? not much point in doing stuff like this when we're not certain how it works

my guess is it's consistent, but would be better to know for sure before we start adding stuff like this in

Reply:
alright